routing/pathfind_test.go

```diff
@@ -3130,3 +3131,186 @@ func dbFindPath(graph *channeldb.ChannelGraph,
+		// we don't need them for intermediate nodes.
+		aliceBobEdge = &channeldb.CachedEdgePolicy{
+			ChannelID: 1,
+			// We won't actually use this timelock / fee (since
+			// it's the sender's outbound channel), but we include
+			// it with different values so that the test will trip
+			// up if we were to include the fee/delay.
+			TimeLockDelta: 10,
+			FeeBaseMSat:   50,
+			ToNodePubKey: func() route.Vertex {
+				return route.NewVertex(bobPk)
+			},
+			ToNodeFeatures: tlvFeatures,
+		}
+
+		bobCarolEdge = &channeldb.CachedEdgePolicy{
+			ChannelID:     2,
+			TimeLockDelta: 15,
+			FeeBaseMSat:   20,
+			ToNodePubKey: func() route.Vertex {
+				return route.NewVertex(carolPk)
+			},
+			ToNodeFeatures: tlvFeatures,
+		}
+
+		// Create edges for the blinded portion of the route with no
+		// fee values, as would be the case when we construct a route
+		// from blinded-route-generated hop hints.
+		carolDaveEdge = &channeldb.CachedEdgePolicy{
+			ToNodePubKey: func() route.Vertex {
+				return route.NewVertex(davePk)
+			},
+			ToNodeFeatures: tlvFeatures,
+		}
+
+		daveEveEdge = &channeldb.CachedEdgePolicy{
+			ToNodePubKey: func() route.Vertex {
+				return route.NewVertex(evePk)
+			},
+			ToNodeFeatures: tlvFeatures,
+		}
+
+		edges = []*channeldb.CachedEdgePolicy{
+			aliceBobEdge,
+			bobCarolEdge,
+			carolDaveEdge,
+			daveEveEdge,
+		}
+
+		// Create the individual hops that we expect in our route.
+		bobHop = &route.Hop{
+			PubKeyBytes:      route.NewVertex(bobPk),
+			ChannelID:        aliceBobEdge.ChannelID,
+			OutgoingTimeLock: 150,
+			AmtToForward:     110,
+		}
+
+		// The introduction node should not have forwarding amount or
+		// expiry, but it should include both her encrypted data and
+		// the blinding point.
+		carolHop = &route.Hop{
+			PubKeyBytes:      route.NewVertex(carolPk),
+			ChannelID:        bobCarolEdge.ChannelID,
+			OutgoingTimeLock: 0,
+			AmtToForward:     0,
+
+			// Carol's payload should have both her encrypted data
+			// and the blinded key for the blinded section of the
+			// route.
+			EncryptedData: carolEncryptedData,
+			BlindingPoint: blindingPk,
+		}
+
+		// Blinded hops should have their encrypted data included, and
+		// the final hop should have our metadata field and amount to
+		// forward / expiry.
+		daveHop = &route.Hop{
+			PubKeyBytes:      route.NewVertex(davePk),
+			OutgoingTimeLock: 0,
+			AmtToForward:     0,
+			EncryptedData:    daveEncryptedData,
+		}
+
+		eveHop = &route.Hop{
+			PubKeyBytes:      route.NewVertex(evePk),
+			OutgoingTimeLock: 150,
+			AmtToForward:     110,
+			EncryptedData:    eveEncryptedData,
+			// Eve's hop should contain fields intended for the
+			// final hop, like metadata.
+			Metadata: metadata,
+		}
+
+		expectedRoute = &route.Route{
+			SourcePubKey: sourceVertex,
+			Hops: []*route.Hop{
+				bobHop,
+				carolHop,
+				daveHop,
+				eveHop,
+			},
+			// Current height = 100
+			// + 50 (blinded portion)
```

Comment:
Nit: Thats not really the blinded portion, thats the final hop which is different from the blinded portion which is 0 in your testcase